libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -902,8 +901,6 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 					solAssert(functionType->kind() == FunctionType::Kind::Internal || functionType->kind() == FunctionType::Kind::DelegateCall, "");
 		}
 	}
-	else
-		solAssert(!functionType->bound(), "");
```

Comment:
Does this fail? AFAIK the only way to access a bound function is via member access. Otherwise the type does not identify itself as bound.

Reply:
Again - `(x.f)() `

Reply:
You got me here.